pkg/controller/operandconfig/operandconfig_controller.go

```diff
@@ -125,5 +170,172 @@ func (r *ReconcileOperandConfig) Reconcile(request reconcile.Request) (reconcile
+		if err != nil {
+			klog.Errorf("Failed to convert alm-examples in the Subscription %s to slice: %s", sub.Name, err)
+			return err
+		}
+
+		merr := &util.MultiErr{}
+
+		// Merge OperandConfig and ClusterServiceVersion alm-examples
+		for _, crTemplate := range crTemplates {
+
+			// Create an unstruct object for CR and request its value to CR template
+			var unstruct unstructured.Unstructured
+			unstruct.Object = crTemplate.(map[string]interface{})
+
+			kind := unstruct.Object["kind"].(string)
+
+			service := instance.GetService(op.Name)
+
+			existinConfig := false
+			for crName := range service.Spec {
+				// Compare the name of OperandConfig and CRD name
+				if strings.EqualFold(kind, crName) {
+					existinConfig = true
+				}
+			}
+
+			if !existinConfig {
+				continue
+			}
+
+			name := unstruct.Object["metadata"].(map[string]interface{})["name"].(string)
+
+			getError := r.client.Get(context.TODO(), types.NamespacedName{
+				Name:      name,
+				Namespace: op.Namespace,
+			}, &unstruct)
+
+			if getError != nil && !errors.IsNotFound(getError) {
+				instance.Status.ServiceStatus[op.Name].CrStatus[kind] = operatorv1alpha1.ServiceFailed
+			} else if errors.IsNotFound(getError) {
+				instance.Status.ServiceStatus[op.Name].CrStatus[kind] = operatorv1alpha1.ServiceNotReady
+			} else {
+				instance.Status.ServiceStatus[op.Name].CrStatus[kind] = operatorv1alpha1.ServiceRunning
+			}
+		}
+		if len(merr.Errors) != 0 {
+			return merr
+		}
+	}
+
+	klog.V(2).Info("Updating OperandConfig status")
+
+	instance.UpdateOperandPhase()
+
+	if err := r.client.Status().Update(context.TODO(), instance); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getRequestToConfigMapper(mgr manager.Manager) handler.ToRequestsFunc {
+	return func(object handler.MapObject) []reconcile.Request {
+		mgrClient := mgr.GetClient()
+		opreqInstance := &operatorv1alpha1.OperandRequest{}
+
+		requests := []reconcile.Request{}
+
+		// If the operandrequest has been deleted, reconcile all the OperandConfig in the cluster
+		if err := mgrClient.Get(context.TODO(), types.NamespacedName{Name: object.Meta.GetName(), Namespace: object.Meta.GetNamespace()}, opreqInstance); errors.IsNotFound(err) {
+			configList := &operatorv1alpha1.OperandConfigList{}
+			_ = mgrClient.List(context.TODO(), configList)
+			for _, config := range configList.Items {
+				namespaceName := types.NamespacedName{Name: config.Name, Namespace: config.Namespace}
+				req := reconcile.Request{NamespacedName: namespaceName}
+				requests = append(requests, req)
+			}
+			return requests
+		}
+
+		// If the operandrequest exist, reconcile OperandConfigs specific in the operandrequest instance.
+		for _, request := range opreqInstance.Spec.Requests {
+			namespace := request.RegistryNamespace
+			if request.RegistryNamespace == "" {
+				namespace = opreqInstance.Namespace
+			}
+			namespaceName := types.NamespacedName{Name: request.Registry, Namespace: namespace}
```

Comment:
You can use `registryKey := opreqInstance.GetRegistryKey(request)` to get NamespacedName, this API method will set the default value.

Reply:
Thanks, I update code to use `GetRegistryKey` to get NamespacedName